packages/clerk-js/src/ui/elements/PasswordInput.tsx

```diff
@@ -1,13 +1,48 @@
+import type { ChangeEvent } from 'react';
 import React, { forwardRef } from 'react';
 
+import { useEnvironment } from '../contexts';
 import { descriptors, Flex, Input } from '../customizables';
+import { usePasswordComplexity } from '../hooks';
 import { EyeSlash } from '../icons';
+import { useFormControl } from '../primitives/hooks';
 import type { PropsOfComponent } from '../styledSystem';
 import { IconButton } from './IconButton';
 
-export const PasswordInput = forwardRef<HTMLInputElement, PropsOfComponent<typeof Input>>((props, ref) => {
+type PasswordInputProps = PropsOfComponent<typeof Input> & {
+  complexity?: boolean;
+};
+
+export const PasswordInput = forwardRef<HTMLInputElement, PasswordInputProps>((props, ref) => {
   const [hidden, setHidden] = React.useState(true);
-  const { id, ...rest } = props;
+  const { id, onChange, complexity = false, ...rest } = props;
+  const formControlProps = useFormControl();
+  const {
+    userSettings: { passwordSettings },
+  } = useEnvironment();
+
+  const { setPassword } = usePasswordComplexity(
+    //TODO: Remove this once default values are populated in DB
+    { ...passwordSettings, max_length: passwordSettings.max_length === 0 ? 999 : 0 },
```

Comment:
@panteliselef 🙃  `max_length` is set to 100 in the BE. Let's replace the 999 with 100

Reply:
I will update this :)